test/UtilsTest.js

```diff
@@ -239,216 +240,142 @@ describe('FlowLimit', async () => {
+
+    const getErrorArgs = async (error, isInbound, flowAmount, flowIn, flowOut, flowLimit, test) => {
+        const contract = test.address;
+
+        switch (error) {
+            case 'FlowAmountOverflow': {
+                const flow = isInbound ? await test.flowInAmount() : await test.flowOutAmount();
+                return [flowAmount, BigNumber.from(flow), contract];
+            }
+
+            case 'FlowLimitExceeded': {
+                const flow = isInbound ? flowIn.add(flowAmount) : flowOut.add(flowAmount);
+                const reverseFlow = isInbound ? flowOut : flowIn;
+                const netFlow = flow.sub(reverseFlow).abs();
+                return [flowLimit, netFlow, contract];
+            }
+
+            case 'FlowAmountExceededLimit':
+                return [flowLimit, flowAmount, contract];
+
+            default:
+                throw new Error(`Unknown error type passed to getErrorArgs: ${error}`);
+        }
+    };
+
+    const executeTestCases = (testCases, expectedError) => {
+        // Flipped test cases should produce the same expected behavior
+        const allCases = [
+            ...testCases,
+            ...testCases.map(({ flowLimit, flows }) => ({
+                flowLimit,
+                flows: flipFlows(flows),
+            })),
+        ];
+
+        for (const { flowLimit, flows } of allCases) {
+            const label = expectedError
+                ? `Should revert with ${expectedError} (flowLimit=${flowLimit}, steps=${flows.length})`
+                : `Should allow valid flow patterns (flowLimit=${flowLimit}, steps=${flows.length})`;
+
+            it(label, async () => {
+                await nextEpoch();
+                await test.setFlowLimit(flowLimit).then((tx) => tx.wait());
+
+                let flowIn = BigNumber.from(0);
+                let flowOut = BigNumber.from(0);
+                const lastIndex = flows.length - 1;
+
+                for (const [index, flow] of flows.entries()) {
+                    const isInbound = flow.in;
+                    const flowAmount = BigNumber.from(flow.in || flow.out);
+                    const fn = isInbound ? test.addFlowIn : test.addFlowOut;
+
+                    if (!expectedError || index < lastIndex) {
+                        await fn(flowAmount).then((tx) => tx.wait());
+
+                        if (isInbound) {
+                            flowIn = flowIn.add(flowAmount);
+                        } else {
+                            flowOut = flowOut.add(flowAmount);
+                        }
+                    } else {
+                        const expectedArgs = await getErrorArgs(expectedError, isInbound, flowAmount, flowIn, flowOut, flowLimit, test);
+                        await expectRevert((gasOptions) => fn(flowAmount, gasOptions), test, expectedError, expectedArgs);
+                    }
+                }
+            });
+        }
+    };
+
+    describe('FlowLimit - Valid flows', () => {
```

Comment:
```suggestion
    describe('Should succeed on valid flows', () => {
```
use consistent naming

Reply:
updated